docs/dev/howtos/development-with-android-components.md

```diff
@@ -0,0 +1,82 @@
+which might end up in failures.
+You can customize which targets are built in `glean/local.properties`
+([more information](https://github.com/ncalexan/rust-android-gradle/blob/master/README.md#specifying-local-targets)):
+
+```
+# For physical devices:
+rust.targets=arm
+
+# For unit tests:
+# rust.targets=darwin # Or linux-*, windows-* (* = x86/x64)
+
+# For emulator only:
+rust.targets=x86
+```
+
+## Substituting projects
+
+android-components has custom build logic for dealing with composite builds,
+so you should be able to configure it by simply adding the path to the Glean repo in the correct `local.properties` file:
+
+In `android-components/local.properties`:
+```groovy
+substitutions.glean.dir=../glean
+```
+
+If this doesn't seem to work, or if you need to configure composite builds for a project that does not contain this custom logic,
+add the following to `settings.gradle`:
+
+In `android-components/settings.gradle`:
+```groovy
+includeBuild('../glean') {
+  dependencySubstitution {
+    substitute module('org.mozilla.telemetry:glean') with project(':glean-core')
+    substitute module('org.mozilla.telemetry:glean-forUnitTests') with project(':glean-core')
+  }
+}
+```
+
+## Caveat
+
+There's a big gotcha with library substitutions: the Gradle build computes lazily, and AARs don't include their transitive dependencies' JNI libraries.
+This means that in `android-components`, `./gradlew :service-glean:assembleDebug` **does not** invoke `:glean-core:cargoBuild`,
+even though `:service-glean` depends on the substitution for `:glean-core` and even if the inputs to Cargo have changed!
+It's the final consumer of the `:service-glean` project (or publication) that will incorporate the JNI libraries.
+
+In practice that means _you should always be targeting something that produces an APK_: a test or a sample module.
+Then you should find that the `cargoBuild` tasks are invoked as you expect.
+
+`./gradlew :samples-glean:connectedAndroidTest` should work.  
+Other tests like `:glean-core:testDebugUnitTest` or `:support-sync-telemetry:testDebugUnitTest` will currently fail, because the JNI libraries are not included.
+
+
+## Notes
+
+This document is based on the equivalent documentation for application-services:
+[Development with the Reference Browser](https://github.com/mozilla/application-services/blob/master/docs/howtos/working-with-reference-browser.md)
+
+1. Transitive substitutions (as shown above) work but require newer Gradle versions (4.10+).
+1. Android Studio happily imports substitutions, but it doesn't appear to always do the right thing with transitive substitutions.  
+   Best to keep substitutions in the final project (i.e., in `reference-browser/settings.gradle`) or to duplicate them in all transitive links.
+1. Be aware that the project list can get very large!  At this time, there's no way to filter the project list.
```

Comment:
What is the "project list" in this context?

Reply:
Can be removed, turns out the `:glean` projects list isn't included in anything visible.